Src/Plugins/Security/SEC101_038.PostgreSqlConnectionStringValidator.cs

```diff
@@ -35,18 +35,14 @@ static PostgreSqlConnectionStringValidator()
             RegexEngine = RE2Regex.Instance;
         }
 
-        public static ValidationState IsValidStatic(ref string matchedPattern,
-                                                    ref Dictionary<string, string> groups,
-                                                    ref string message,
-                                                    out ResultLevelKind resultLevelKind,
-                                                    out Fingerprint fingerprint)
+        public static IEnumerable<ValidationResult> IsValidStatic(ref string matchedPattern,
+                                                                  ref Dictionary<string, string> groups,
+                                                                  ref string message)
```

Comment:
>message [](http://example.com/codeflow?start=77&length=7)

message needs to be a part of validation result. i.e., it's not a given that there will be a single message only for each validation result. 

generally our pattern for the static validator: all input args are to process against, all return values are bundled in ValidationResult.

by the way! i don't know why we bothered to make Dictionary<string, string> a 'ref' arg! That doesn't make any sense. You can still mutate the groups data in arbitrary ways by adding keys, or clearing the dictionary. There's no need to replace the dictionary entirely.